From: Jiri Slaby <jslaby@suse.cz>
Date: Thu, 30 Mar 2017 10:52:48 +0200
Subject: drm/i915: disable KASAN for handlers
Patch-mainline: submitted, https://lkml.kernel.org/r/<20170330094627.29460-1-jslaby@suse.cz>
References: bnc#1025903

Handlers are currently the only blocker to compile the kernel with gcc 7
and KASAN+use-after-scope enabled:
drivers/gpu/drm/i915/gvt/handlers.c:2200:1: error: the frame size of 43760 bytes is larger than 2048 bytes [-Werror=frame-larger-than=]
drivers/gpu/drm/i915/gvt/handlers.c:2402:1: error: the frame size of 9400 bytes is larger than 2048 bytes [-Werror=frame-larger-than=]
drivers/gpu/drm/i915/gvt/handlers.c:2628:1: error: the frame size of 11256 bytes is larger than 2048 bytes [-Werror=frame-larger-than=]

It is due to many expansions of MMIO_* macros in init_generic_mmio_info.
INTEL_GVT_MMIO_OFFSET generates for each such line a __reg and an
offset. There are too many for KASAN to keep up.

So disable KASAN for this file.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Cc: Martin Liska <mliska@suse.cz>
Cc: Zhenyu Wang <zhenyuw@linux.intel.com>
Cc: Zhi Wang <zhi.a.wang@intel.com>
Cc: Daniel Vetter <daniel.vetter@intel.com>
Cc: Jani Nikula <jani.nikula@linux.intel.com>
Cc: David Airlie <airlied@linux.ie>
Cc: intel-gvt-dev@lists.freedesktop.org
Cc: intel-gfx@lists.freedesktop.org
Cc: dri-devel@lists.freedesktop.org
---
 drivers/gpu/drm/i915/gvt/Makefile |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/gpu/drm/i915/gvt/Makefile
+++ b/drivers/gpu/drm/i915/gvt/Makefile
@@ -6,3 +6,5 @@ GVT_SOURCE := gvt.o aperture_gm.o handle
 ccflags-y				+= -I$(src) -I$(src)/$(GVT_DIR) -Wall
 i915-y					+= $(addprefix $(GVT_DIR)/, $(GVT_SOURCE))
 obj-$(CONFIG_DRM_I915_GVT_KVMGT)	+= $(GVT_DIR)/kvmgt.o
+
+KASAN_SANITIZE_handlers.o := n
